### ISSUE TITLE
  Homestead.yaml Folder in Vagrant not created (SAME ERROR)

### ISSUE THREAD

**Issue by yupthatguy** [laravel/homestead#847]
Please note that the Homestead issue tracker is reserved for bug reports and enhancements. We are not always able to debug Vagrant, Provider or Operating System issues, but will do our best to help. Thank you!

## Versions
  - Vagrant: Vagrant 1.9.7
  - Provider: Virtualbox Version 5.1.34_Ubuntu r121010  (installed / maintained via synatic)
  - Homestead: `homestead --version`  will not show version `homestead: command not found` , but I am running v6.0.0-alpha1

**A lot of issues can be resolved by simply updating vagrant, provider or homestead.**
>**Note:** Virtualbox users, please upgrade to `~5.2`. You *will* show as up-to-date from the `~5.0` version when you `About -> Check for Updates`. You'll need to download a newer version of [Virtualbox](https://www.virtualbox.org/wiki/Downloads).

## Host operating system
Ubuntu 16.04

## Homestead.yaml
```yaml
---
ip: "192.168.10.10"
memory: 2048
cpus: 1
provider: virtualbox

authorize: ~/.ssh/id_rsa.pub

keys:
    - ~/.ssh/id_rsa

folders:
    - map: ~/code/site1
      to: /home/user/VBShared/public_html/site1
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site2
      to: /home/user/VBShared/public_html/site2
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site3
      to: /home/user/VBShared/public_html/site3
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site4
      to: /home/user/VBShared/public_html/site4
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site5
      to: /home/user/VBShared/public_html/site5
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site6
      to: /home/user/VBShared/public_html/site6
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site7
      to: /home/user/VBShared/public_html/site7
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site8
      to: /home/user/VBShared/public_html/site8
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site9
      to: /home/user/VBShared/public_html/site9
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site10
      to: /home/user/VBShared/public_html/site10
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site11
      to: /home/user/VBShared/public_html/site11
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site12
      to: /home/user/VBShared/public_html/site12
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

    - map: ~/code/site13
      to: /home/user/VBShared/public_html/site13
      type: "rsync"
      options:
          rsync__args: ["-v", "-a", "-h", "-r", "-u", "--delete", "-z"]

sites:
    
    - map: site1
      to: /home/user/VBShared/public_html/site1
      type: "apache"
      php: "7.1"

    - map: site2
      to: /home/user/VBShared/public_html/site2
      type: "apache"

    - map: site3
      to: /home/user/VBShared/public_html/site3
      type: "apache"
      php: "7.1"

    - map: site4
      to: /home/user/VBShared/public_html/site4
      type: "apache"
      php: "7.1"

    - map: site5
      to: /home/user/VBShared/public_html/site5
      type: "apache"
      php: "7.1"

    - map: site6
      to: /home/user/VBShared/public_html/site6
      type: "apache"
      php: "7.1"

    - map: site7
      to: /home/user/VBShared/public_html/site7
      type: "apache"
      php: "7.1"

    - map: site8
      to: /home/user/VBShared/public_html/site8
      type: "apache"
      php: "7.1"

    - map: site9
      to: /home/user/VBShared/public_html/site9
      type: "apache"
      php: "7.1"

    - map: site10
      to: /home/user/VBShared/public_html/site10
      type: "apache"
      php: "7.1"

    - map: site11
      to: /home/user/VBShared/public_html/site11
      type: "apache"
      php: "7.1"

    - map: site12
      to: /home/user/VBShared/public_html/site12
      type: "apache"
      php: "7.1"

    - map: site13
      to: /home/user/VBShared/public_html/site13
      type: "apache"
      php: "7.1"

databases:
    - homestead

# blackfire:
#     - id: foo
#       token: bar
#       client-id: foo
#       client-token: bar

# ports:
#     - send: 50000
#       to: 5000
#     - send: 7777
#       to: 777
#       protocol: udp
```

## Vagrant destroy & up output

`==> homestead-7: Unable to mount one of your folders. Please check your folders in Homestead.yaml`

## Expected behavior

The Folder "code" in vagrant is not created

## Actual behavior

1.) When I ran `vagrant reload --provision`  I was told that my version of virtualbox did NOT match my installed guestadditions.iso.

2.) Therefore, I ran `vagrant plugin install vagrant-vbguest` and rebooted my machine

3.)  However, now when I run `vagrant reload --provision`  ... I no longer receive an error message about mismatching vbox & guestadditions..... but I still get the error `==> homestead-7: Unable to mount one of your folders. Please check your folders in Homestead.yaml`

4a.) Also, maybe related, maybe not  `$homestead --version does not work`

4b.) Also, maybe related, maybe not `==> homestead-7: You are running composer as "root", while "/home/vagrant/.composer" is owned by "vagrant"

`==> homestead-7: You are already using composer version 1.6.4 (stable channel).`

## Steps to reproduce
1.  vagrant reload --provision

## References
Are there any other [GitHub issues](https://github.com/laravel/homestead/issues?utf8=%E2%9C%93&q=is%3Aissue) (open or closed) that should be linked here?
For example:
- https://github.com/laravel/homestead/issues/606

**Comment by yupthatguy** [comment]
Admittedly, I am a newbie, but I like your software. It will make Drupal development much easier.  In any case I finally got everything to work (minus email).

this is my final folder structure

- map: ~/VBShared/public_html/site1
      to: /home/user/VBShared/public_html/site1

Just looking for confirmation that the above is correct.   

I am setting up a 2nd development environment as a backup that will share the same public_html folder.  At the end of every session I dump the db into the root directory of whatever drupal site.  Will all of the changes made be written back to the host immediately or would I have to run a "vagrant reload --provision" in order to rsync changes with my host before I can switch to a backup dev environment?

FYI, I just reinstalled virtualbox to get it to match the Homestead requested version and all is well on that front. 

thanks
